src/SystemCommands-MethodCommands/SycPushDownMethodInSomeClassesCommand.class.st

```diff
@@ -0,0 +1,68 @@
+Class {
+	#name : #SycPushDownMethodInSomeClassesCommand,
+	#superclass : #SysRefactoringMethodCommand,
+	#instVars : [
+		'classes'
+	],
+	#category : #'SystemCommands-MethodCommands'
+}
+
+{ #category : #activation }
+SycPushDownMethodInSomeClassesCommand class >> methodContextMenuActivation [
+	<classAnnotation>
+	
+	^ SycMethodMenuActivation byRootGroupItemOrder: 21 for: ClyMethod asCalypsoItemContext
+]
+
+{ #category : #converting }
+SycPushDownMethodInSomeClassesCommand >> asRefactorings [
+	"Return push down method refactoring using method"
+	| refactoring |
+	refactoring := RBPushDownMethodRefactoring 
+		model: model
+		pushDown: (methods collect: [ :each | each selector ])
+		from: methods first origin
+		in: self classes.
+	^ OrderedCollection with: refactoring 
+]
+
+{ #category : #converting }
+SycPushDownMethodInSomeClassesCommand >> classes [
+	^ classes ifNil: [ classes := #() ]
+]
+
+{ #category : #accessing }
+SycPushDownMethodInSomeClassesCommand >> defaultMenuIconName [ 
+	^ #down
+]
+
+{ #category : #accessing }
+SycPushDownMethodInSomeClassesCommand >> defaultMenuItemName [
+	^'Push down in some classes'
+]
+
+{ #category : #execution }
+SycPushDownMethodInSomeClassesCommand >> prepareFullExecutionInContext: aToolContext [
+	super prepareFullExecutionInContext: aToolContext.
+	self selectMethodsAndClasses.
+]
+
+{ #category : #execution }
+SycPushDownMethodInSomeClassesCommand >> selectMethodsAndClasses [
+	|selectedMethods selectedClasses dialog class|
+	class := methods first origin.
+	selectedMethods := methods copy asOrderedCollection.
```

Comment:
I was wondering why do we need to copy the method

Reply:
My mistake, I already remove the use of unnecessary temporary variables